src/coreclr/jit/lsraarm64.cpp

```diff
@@ -2045,22 +2032,38 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
             {
                 prefUseOpNum = 2;
             }
-            GenTree* prefUseNode = embOp2Node->Op(prefUseOpNum);
-            for (size_t argNum = 1; argNum <= numArgs; argNum++)
+            prefUseNode = embOp2Node->Op(prefUseOpNum);
+        }
+
+        for (size_t argNum = 1; argNum <= numArgs; argNum++)
+        {
+            GenTree* node = embOp2Node->Op(argNum);
+
+            if (node == prefUseNode)
+            {
+                tgtPrefUse = BuildUse(node);
+                srcCount++;
+            }
+            else
             {
-                if (argNum == prefUseOpNum)
-                {
-                    tgtPrefUse = BuildUse(prefUseNode);
-                    srcCount += 1;
-                }
-                else
+                RefPosition* useRefPosition = nullptr;
+
+                srcCount += BuildDelayFreeUses(node, nullptr, RBM_NONE, &useRefPosition);
+
+#if defined(DEBUG)
+                // Ensure that if this node and the RMW node refer to the same local variable, then this
+                // node must be marked as delay free.
+                if (isCandidateLocalRef(node) && isCandidateLocalRef(prefUseNode) &&
+                    (getIntervalForLocalVarNode(node->AsLclVar()) ==
+                     getIntervalForLocalVarNode(prefUseNode->AsLclVar())))
                 {
-                    srcCount += BuildDelayFreeUses(embOp2Node->Op(argNum), prefUseNode);
+                    assert(useRefPosition->delayRegFree);
                 }
+#endif // defined(DEBUG)
```

Comment:
Can it be just
```
// It is a hard requirement that these are not allocated to the same register as the destination,
// so verify no optimizations kicked in to skip setting the delay-free.
assert(useRefPosition->delayRegFree);
```
? The requirements apply equally to all operands, not just in the "same local" cases.

Reply:
> ? The requirements apply equally to all operands, not just in the "same local" cases.

Done as suggested. Had to add an check that a use was actually created.